app/controllers/pieces_controller.rb

```diff
@@ -12,10 +12,16 @@ def update
       flash[:danger] = "#{@piece.color.capitalize} #{@piece.type} move to X#{x}/Y#{y} was invalid"
     end
 
+    if @piece.game.en_passant_opportunity_active?
+      @piece.game.update_attribute(:allows_en_passant_capture?, true)
+    else
+      @piece.game.update_attribute(:allows_en_passant_capture?, false) if @piece.game.allows_en_passant_capture?
+    end
+
```

Comment:
Alright, finally some time to respond :)

Yeah, I wouldn't be surprised if there was a better, 'more appropriate' way to handle this logic as far as where to put it and what have you. I don't know if there is, but I'm happy to look at it. Might have an opportunity this weekend, we'll see.

If it makes sense to move it into the Game model, I'll give it a go. Don't know what would or wouldn't work at the moment.